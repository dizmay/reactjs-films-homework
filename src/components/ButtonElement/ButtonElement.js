import React from 'react';
import PropTypes from 'prop-types';
import styles from './ButtonElement.scss';

const ButtonElement = ({ children, buttonStyle }) => (
  <button type="button" className={`${styles.btn} ${buttonStyle ? styles.blue : styles.transparent}`}>{children}</button>
);

ButtonElement.defaultProps = {
  children: null,
  buttonStyle: true,
};

ButtonElement.propTypes = {
  children: PropTypes.string,
  buttonStyle: PropTypes.bool,
};

export default ButtonElement;
