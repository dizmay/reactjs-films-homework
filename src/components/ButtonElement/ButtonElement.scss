.btn {
  width: 9rem;
  height: 3rem;
  border-radius: 0.3rem;
  outline: none;
  cursor: pointer;
  color: #fff;
  font-size: 1.2rem;
}

.blue {
  background: #03c6fc;
  border: 0.2rem solid #00b2e3;
}

.transparent {
  background: transparent;
  border: 0.2rem solid #fff;
}
