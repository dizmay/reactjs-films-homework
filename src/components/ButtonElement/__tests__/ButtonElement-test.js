import React from 'react';
import renderer from 'react-test-renderer';
import ButtonElement from '../ButtonElement';

it('renders correctly', () => {
  const tree = renderer.create(<ButtonElement />).toJSON();
  expect(tree).toMatchSnapshot();
});
