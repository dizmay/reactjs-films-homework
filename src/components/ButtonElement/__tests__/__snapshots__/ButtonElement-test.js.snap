// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="btn blue"
  type="button"
/>
`;
